driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/dup3.bpf.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2022 The Falco Authors.
+ *
+ * This file is dual licensed under either the MIT or GPL 2. See MIT.txt
+ * or GPL2.txt for full copies of the license.
+ */
+
+#include <helpers/interfaces/fixed_size_event.h>
+
+/*=============================== ENTER EVENT ===========================*/
+
+SEC("tp_btf/sys_enter")
+int BPF_PROG(dup3_e,
+	     struct pt_regs *regs,
+	     long id)
+{
+	struct ringbuf_struct ringbuf;
+	if(!ringbuf__reserve_space(&ringbuf, DUP3_E_SIZE))
+	{
+		return 0;
+	}
+
+	ringbuf__store_event_header(&ringbuf, PPME_SYSCALL_DUP3_E, DUP3_E_SIZE);
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+
+	/* Parameter 1: fd (type: PT_FD) */
+	s32 fd = (s32)extract__syscall_argument(regs, 0);
+	ringbuf__store_s64(&ringbuf, (s64)fd);
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+
+	ringbuf__submit_event(&ringbuf);
+
+	return 0;
+}
+
+/*=============================== ENTER EVENT ===========================*/
+
+/*=============================== EXIT EVENT ===========================*/
+
+SEC("tp_btf/sys_exit")
+int BPF_PROG(dup3_x,
+	     struct pt_regs *regs,
+	     long ret)
+{
+	struct ringbuf_struct ringbuf;
+	if(!ringbuf__reserve_space(&ringbuf, DUP3_X_SIZE))
+	{
+		return 0;
+	}
+
+	ringbuf__store_event_header(&ringbuf, PPME_SYSCALL_DUP3_X, DUP3_X_SIZE);
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+
+	/* Parameter 1: res (type: PT_FD)*/
+	ringbuf__store_s64(&ringbuf, ret);
+
+	/* Parameter 2: oldfd (type: PT_FD) */
+	s32 oldfd = (s32)extract__syscall_argument(regs, 0);
+	ringbuf__store_s64(&ringbuf, (s64)oldfd);
+
+	/* Parameter 3: newfd (type: PT_FD) */
+	s32 newfd = (s32)extract__syscall_argument(regs, 1);
+	ringbuf__store_s64(&ringbuf, (s64)newfd);
+
+	/* Parameter 4: flags (type: PT_FLAGS32) */
+	unsigned long flags = extract__syscall_argument(regs, 2);
```

Comment:
Question: looking at system call dup3() man page, the flags of type `int`. Any reason to use unsigned here?

Reply:
uhm that's a good point, don't know why, I cannot correct it here for the same reason as before, I have to follow the old definitions :/ but we can open an issue for that, great catch!

Reply:
thinking again about it... it could be ok if we convert the `int` to `uint32_t` the important thing is that our internal representation `PPM_...` is always positive. By the way, there are cases in which we catch the `int` value and push it to userspace as a `uint32_t` without performing any conversion :/